pkg/server/server.go

```diff
@@ -277,28 +280,51 @@ func (s *Server) Run(ctx context.Context) error {
 		return err
 	}
 
-	// FIXME: switch to a single set of shared informers
 	const crossCluster = "*"
+
+	// Setup kcp * informers
 	kcpClusterClient, err := kcpclient.NewClusterForConfig(loopbackClientConfig)
 	if err != nil {
 		return err
 	}
 	kcpClient := kcpClusterClient.Cluster(crossCluster)
+	s.kcpSharedInformerFactory = kcpexternalversions.NewSharedInformerFactoryWithOptions(kcpClient, resyncPeriod)
+
+	// Setup kube * informers
+	kubeClusterClient, err := kubernetes.NewClusterForConfig(loopbackClientConfig)
+	if err != nil {
+		return err
+	}
+	kubeClient := kubeClusterClient.Cluster(crossCluster)
+	s.kubeSharedInformerFactory = coreexternalversions.NewSharedInformerFactoryWithOptions(kubeClient, resyncPeriod)
 
-	kcpSharedInformerFactory := kcpexternalversions.NewSharedInformerFactoryWithOptions(kcpClient, resyncPeriod)
-	s.AddPostStartHook("FIXME-start-informers-for-crd-getter", func(context genericapiserver.PostStartHookContext) error {
+	// Setup apiextensions * informers
+	apiextensionsClusterClient, err := apiextensionsclient.NewClusterForConfig(loopbackClientConfig)
+	if err != nil {
+		return err
+	}
+	apiextensionsClient := apiextensionsClusterClient.Cluster(crossCluster)
+	s.apiextensionsSharedInformerFactory = apiextensionsexternalversions.NewSharedInformerFactoryWithOptions(apiextensionsClient, resyncPeriod)
+
+	s.AddPostStartHook("start-informers", func(context genericapiserver.PostStartHookContext) error {
 		if err := s.waitForCRDServer(context.StopCh); err != nil {
 			return err
 		}
-		kcpSharedInformerFactory.Start(context.StopCh)
-		kcpSharedInformerFactory.WaitForCacheSync(context.StopCh)
+
+		s.kcpSharedInformerFactory.Start(context.StopCh)
+		s.kubeSharedInformerFactory.Start(context.StopCh)
+		s.apiextensionsSharedInformerFactory.Start(context.StopCh)
+
+		s.kcpSharedInformerFactory.WaitForCacheSync(context.StopCh)
```

Comment:
handle errors. This is not enough.

Reply:
Or do we know these run indefinitely?

Reply:
They should go indefinitely. They call cache.WaitForCacheSync underneath